adal/src/Microsoft.IdentityModel.Clients.ActiveDirectory/Internal/Flows/AcquireDeviceCodeHandler.cs

```diff
@@ -84,15 +84,30 @@ private string CreateDeviceCodeRequestUriString()
 
                 deviceCodeRequestParameters.ExtraQueryParameter = extraQueryParameters;
             }
+            return deviceCodeRequestParameters;
+        }
+
+        private string CreateDeviceCodeRequestUriString()
+        {
+            DictionaryRequestParameters deviceCodeRequestParameters = GetRequestParameters();
 
             return new Uri(new Uri(this.authenticator.DeviceCodeUri), "?" + deviceCodeRequestParameters).AbsoluteUri;
         }
 
         internal async Task<DeviceCodeResult> RunHandlerAsync()
         {
             await this.authenticator.UpdateFromTemplateAsync(this.requestContext).ConfigureAwait(false);
-            this.ValidateAuthorityType();
-            AdalHttpClient client = new AdalHttpClient(CreateDeviceCodeRequestUriString(), this.requestContext);
+            AdalHttpClient client;
+            if (authenticator.AuthorityType == AuthorityType.ADFS)
+            {
+                //Adfs devicecode endpoint only supports POST requests
+                client = new AdalHttpClient(authenticator.DeviceCodeUri, this.requestContext);
+                client.Client.BodyParameters = GetRequestParameters();
```

Comment:
>client.Client.BodyParameters = GetRequestParameters(); [](start = 16, length = 54)

Mark: can we flip all our requests to Post (for Device Code)?

Reply:
AAD supports both Get and Post, so that should be fine